src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -45,10 +49,17 @@ public SubstringConstraint(string expected) : base(expected)
         /// <returns>True for success, false for failure</returns>
         protected override bool Matches(string actual)
         {
-            if (this.caseInsensitive)
-                return actual != null && actual.ToLower().IndexOf(expected.ToLower()) >= 0;
-            else
-                return actual != null && actual.IndexOf(expected) >= 0;
+            return actual != null && actual.IndexOf(expected, comparisonType) >= 0;
+        }
+
+        /// <summary>
+        /// Overrides the <see cref="StringConstraint.IgnoreCase"/> setting and instead uses the specified comparison.
+        /// </summary>
+        public SubstringConstraint Using(StringComparison? comparisonType)
+        {
+            this.comparisonType = comparisonType ?? (caseInsensitive ? 
+                StringComparison.CurrentCultureIgnoreCase : StringComparison.Ordinal);
```

Comment:
@MhdTlb I apologize! In the comment I cc'd you on, I realized that we need `StringComparison.CurrentCultureIgnoreCase : StringComparison.CurrentCulture` to match the current behavior because in the old code, the `IndexOf` was actually doing a `CurrentCulture` comparison.